obspy/io/gse2/libgse2.py

```diff
@@ -331,6 +331,26 @@ def verify_checksum(fh, data, version=2):
     return
 
 
+def read_integer_data(fh, npts):
+    """
+    Reads npts points of uncompressed integers from given file handler.
+    """
+    # find next DAT2 section within file
+    data = []
+    in_data_section = False
+
+    while len(data) < npts:
```

Comment:
You need to also check if `fh` is exhausted, or you will get an infinite loop if the file is truncated.